tests/integration/api/test_search_format.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+import requests
+from urllib.parse import quote
+from .constants import API_URL
+
+@pytest.mark.parametrize("endpoint", [
+    ("/search?query=keyword:NASA&filter=format:readable")
+])
+def test_readable_format_flags(endpoint):
+    url = API_URL + quote(endpoint)
+    response = requests.get(url)
+    
+    response_json = response.json()
+    assert response_json is not None, "Response JSON is empty"
+    
+    works = response_json.get('data', {}).get('works', [])
+     
+    for work in works:
+        for edition in work.get('editions', []):
+            for item in edition.get('items', []):
+                if any(link.get('mediaType') == "text/html" for link in item.get('links', [])):
+                    flags = item.get('links', [])[0].get('flags', {})
+                    assert flags.get('reader', False) or flags.get('embed', False), \
+                        f"Flags validation failed: {flags}"
```

Comment:
Let's pull the html links out here and check to make sure that the reader flag is false and the embed is true for each one. 

The logic you have here checks if any link has a media type of text/html but then grabs the first link's flags. However, the link at that index may not have a mediaType of "text/html". 

You could potentially go further here and check other media type and ensure that the reader flag is True (I think).